CMakeLists.txt

```diff
@@ -48,16 +48,78 @@ if ((FIPS_MACOS OR FIPS_WINDOWS OR FIPS_LINUX) AND NOT (FIPS_RASPBERRYPI OR FIPS
+                find_package(ECM REQUIRED NO_MODULE)
+                list(APPEND CMAKE_MODULE_PATH "${ECM_MODULE_PATH}" "${CMAKE_SOURCE_DIR}/cmake_modules")
+
+                find_package(Wayland REQUIRED Client Cursor Egl)
+                find_package(WaylandScanner REQUIRED)
+                find_package(WaylandProtocols 1.15 REQUIRED)
+
+                include_directories("${Wayland_INCLUDE_DIRS}")
+                fips_libs("${Wayland_LIBRARIES}" "${CMAKE_THREAD_LIBS_INIT}" EGL)
+
+                find_package(XKBCommon REQUIRED)
+                include_directories("${XKBCOMMON_INCLUDE_DIRS}")
+
+                include(CheckIncludeFiles)
+                include(CheckFunctionExists)
+                check_include_files(xkbcommon/xkbcommon-compose.h HAVE_XKBCOMMON_COMPOSE_H)
+                check_function_exists(memfd_create HAVE_MEMFD_CREATE)
+                ecm_add_wayland_client_protocol(glfw_SOURCES
+                    PROTOCOL
+                    "${WAYLAND_PROTOCOLS_PKGDATADIR}/stable/xdg-shell/xdg-shell.xml"
+                    BASENAME xdg-shell)
+                ecm_add_wayland_client_protocol(glfw_SOURCES
+                    PROTOCOL
+                    "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/xdg-decoration/xdg-decoration-unstable-v1.xml"
+                    BASENAME xdg-decoration)
+                ecm_add_wayland_client_protocol(glfw_SOURCES
+                    PROTOCOL
+                    "${WAYLAND_PROTOCOLS_PKGDATADIR}/stable/viewporter/viewporter.xml"
+                    BASENAME viewporter)
+                ecm_add_wayland_client_protocol(glfw_SOURCES
+                    PROTOCOL
+                    "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/relative-pointer/relative-pointer-unstable-v1.xml"
+                    BASENAME relative-pointer-unstable-v1)
+                ecm_add_wayland_client_protocol(glfw_SOURCES
+                    PROTOCOL
+                    "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/pointer-constraints/pointer-constraints-unstable-v1.xml"
+                    BASENAME pointer-constraints-unstable-v1)
+                ecm_add_wayland_client_protocol(glfw_SOURCES
+                    PROTOCOL
+                    "${WAYLAND_PROTOCOLS_PKGDATADIR}/unstable/idle-inhibit/idle-inhibit-unstable-v1.xml"
+                    BASENAME idle-inhibit-unstable-v1)
+                set(CurDir "")
+                fips_files("${glfw_SOURCES}")
+                include_directories("${CMAKE_CURRENT_BINARY_DIR}")
+                fips_dir(glfw/src GROUP .)
+                fips_files(
+                    wl_platform.h
+                    posix_time.h posix_thread.h xkb_unicode.h egl_context.h
+                    osmesa_context.h
+                    wl_init.c wl_monitor.c wl_window.c
+                    posix_time.c posix_thread.c xkb_unicode.c
+                    egl_context.c osmesa_context.c
+                    linux_joystick.h linux_joystick.c
+                )
+                fips_libs(GL)
```

Comment:
I had to change the linker dependencies to include the math and dl libs (same as in the non-Wayland path):

```fips_libs(GL m ${CMAKE_DL_LIBS})```

I guess in Oryol this didn't trigger because those libs are added elsewhere.